presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1397,5 +1406,10 @@ private boolean isCandidate(Map<ColumnHandle, NullableValue> bindings)
 
             return true;
         }
+
+        private <T> boolean intersects(Set<T> one, Set<T> other)
+        {
+            return intersection(one, other).stream().findFirst().isPresent();
```

Comment:
`! intersection(one, other).isEmpty() `?

Reply:
no,  `isEmpty()` is implemented with `size() == 0` I do not care the size

Reply:
Did you mean `com.google.common.collect.Sets.SetView#isEmpty` ?